File: app/src/main/java/com/xmx/tango/Tools/Data/SQL/InsertCallback.java
package com.xmx.tango.Tools.Data.SQL;

import com.avos.avoscloud.AVException;
import com.avos.avoscloud.AVObject;

/**
 * Created by The_onE on 2016/1/11.
 */
public abstract class InsertCallback {

    public abstract void proceeding(int index);

    public abstract void success(int total);
}

File: app/src/main/java/com/xmx/tango/common/data/sql/InsertCallback.kt
package com.xmx.tango.common.data.sql

/**
 * Created by The_onE on 2016/1/11.
 * 批量插入回调
 */
abstract class InsertCallback {

    /**
     * 每次插入后执行的操作
     * @param index 已插入的数量
     */
    abstract fun proceeding(index: Int)

    /**
     * 全部插入成功后执行的操作
     * @param total 插入的总数
     */
    abstract fun success(total: Int)

    /**
     * 中途执行失败的操作
     * @param e 异常信息
     */
    abstract fun error(e: Exception)
}
